sky/authentication.py

```diff
@@ -1,20 +1,25 @@
 """Module to enable a single SkyPilot key for all VMs in each cloud.
 
-The `setup_<cloud>_authentication` functions will be called after the ray
-yaml template file `<cloud>-ray.yml.j2` is filled in with resource specific
-information. The functions will take the filled in ray yaml config as input,
+The `setup_<cloud>_authentication` functions will be called on every cluster
+provisioning request.
+
+Specifically, after the ray yaml template file `<cloud>-ray.yml.j2` is filled in
+with resource specific information, these functions are called with the filled
+in ray yaml config as input, yaml template file `<cloud>-ray.yml.j2` is filled
+in with resource specific information. The functions will take the filled in ray
+yaml config as input,
```

Comment:
some duplication

Reply:
Oops, the applying of the changes was buggy. Fixed. Thanks!